### ISSUE TITLE
[a11y]: <BackButton /> isn't keyboard/screen reader friendly

### ISSUE THREAD

**Issue by st0nebraker** [sourcegraph/about#5627]
### Audit type

Keyboard navigation

### User journey audit issue

#5467

### Problem description

The hero's `<BackButton />` is not focusable on keyboard navigation. Additionally, screen readers don't read them as links, so it's unclear to the user what the element can do.

Uploading Screen Recording 2022-07-29 at 9.57.08 AM.mov…

### Expected behavior

The `<BackButton />` should probably have button tag so that it can be focusable and read appropriately by screen readers.

**Comment by felixfbecker** [comment]
@st0nebraker looks like the recording didn't upload, do you still have the recording ("Screen Recording 2022-07-29 at 9.57.08 AM.mov")? Not sure what this issue refers to :)

**Comment by st0nebraker** [comment]
Woops, my bad! I actually don't see this bug anymore, it must've been fixed since July 29th this year. Closing the issue.
